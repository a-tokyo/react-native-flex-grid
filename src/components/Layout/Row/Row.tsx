import React from 'react';
import { StyleSheet, View, ViewProps } from 'react-native';

import { getConfig } from '../../../utils/grid';

export declare interface RowProps extends ViewProps {
  /** Gutter size -- [Bootstrap Gutters](https://getbootstrap.com/docs/5.0/layout/gutters/) */
  gx?: 0 | 1 | 2 | 3 | 4 | 5;
  /** Direction */
  dir?: 'ltr' | 'rtl';
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  rtl: {
    flexDirection: 'row-reverse',
  },
});

/** Row */
const Row = ({ style, gx = 4, dir = 'ltr', ...rest }: RowProps) => (
  <View
    style={[
      styles.row,
      dir === 'rtl' && styles.rtl,
      gx
        ? {
            marginHorizontal: (-1 * getConfig().gutters[gx]) / 2,
          }
        : undefined,
      style,
    ]}
    {...rest}
  />
);

export default Row;
