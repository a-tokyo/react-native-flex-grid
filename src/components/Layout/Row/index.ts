export * from './Row';

export { default } from './Row';
