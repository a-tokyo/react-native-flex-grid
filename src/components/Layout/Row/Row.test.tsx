import React from 'react';
import renderer from 'react-test-renderer';
import Row from './Row';

const setupRow = (overrideProps = {}) => <Row {...overrideProps} />;

describe('<Row />', () => {
  it('renders basic props correctly', () => {
    const wrapper = renderer.create(setupRow());

    expect(wrapper).toMatchSnapshot();
  });
});
