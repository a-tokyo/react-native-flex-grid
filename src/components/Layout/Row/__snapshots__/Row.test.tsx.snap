// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> renders basic props correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
      },
      false,
      {
        "marginHorizontal": -12,
      },
      undefined,
    ]
  }
/>
`;
